### ISSUE TITLE
Landing page + automatic rewrite of OpenGrok URLs

### ISSUE THREAD

**Issue by mike-r-mclaughlin** [sourcegraph/sourcegraph#40230]
#### Feature request description

For users transitioning off of OpenGrok, it would be nice to be able to redirect the old OpenGrok URL to Sourcegraph and have it automatically rewrite the paths to Sourcegraph URLs/queries.  

The general idea is to point the former OpenGrok DNS entry to Sourcegraph and have any previously bookmarked URLs (or generated URLs) convert automatically to Sourcegraph URLs so users will see what they expect to see.

Examples:
| Name                | OpenGrok URL | Sourcegraph URL |
|----------------|-------------------|--------------------------------------|
| Directory          | http://192.168.64.6:8080/source/xref/sourcegraph/client/web/src/enterprise/code-monitoring/testing/ | http://192.168.64.6/github.com/sourcegraph/sourcegraph/-/blob/client/web/src/enterprise/code-monitoring/testing/ |
| File                    | http://192.168.64.6:8080/source/xref/sourcegraph/client/web/src/enterprise/code-monitoring/testing/util.ts?r=b23a28ce | http://192.168.64.6/github.com/sourcegraph/sourcegraph@04cf0cb07327c01141fad5eb9707600c7eb806c2/-/blob/client/web/src/enterprise/code-monitoring/testing/util.ts |
| query: Symbol Search `OverwriteSettings`  | http://192.168.64.6:8080/source/search?full=&defs=OverwriteSettings&refs=&path=&hist=&type=&xrd=&nn=2&si=full&searchall=true&si=full | http://192.168.64.6/search?q=context:global+repo:%5Egithub%5C.com/sourcegraph/sourcegraph%24+type:symbol+OverwriteSettings&patternType=lucky  |
(more to be added)

##### Potential v1

Create rewrite rules that would account for the top ~20 URL patterns.  And, create a method to track the top OpenGrok URL patterns that were not rewritten.  (This will allow us to be agile with adding additional rewrite rules.  This means it must also be easy to add additional rewrite rules.)
[NOTE: this might be as simple as configuring nginx in front of Sourcegraph frontend]

##### Additional nice-to-have features

Create a landing page that the rewritten OpenGrok URLs redirect to first.  That landing page can have a countdown clock on it to stay on it for x seconds (10s for example).  On that landing page, it would show the OpenGrok query on one side and the equivalent Sourcegraph query on the other side and then redirect the user to Sourcegraph with the query populated and executed.

OpenGrok URL -> landing page (shows user how to accomplish same thing in Sourcegraph) -> (delay x seconds) -> Sourcegraph

#### Is your feature request related to a problem? If so, please describe.

When teams transition from OpenGrok to Sourcegraph, it can be challenging to completely turn off OpenGrok because of bookmarked URLs or scripts that generate links to OpenGrok, etc.  Having a redirect/rewrite engine builtin to Sourcegraph would allow these companies to decommission OpenGrok sooner and only have to keep one system up-to-date.  This also leaves less confusion for users since there is only one code search engine.

#### Describe alternatives you've considered.

Implement this fully outside of Sourcegraph

#### Additional context

See this [internal Slack discussion thread](https://sourcegraph.slack.com/archives/C03MV6CKU1X/p1660072424019509) for https://github.com/sourcegraph/accounts/issues/6716

**Comment by mike-r-mclaughlin** [comment]
Additional discussion in [this internal Slack thread](https://sourcegraph.slack.com/archives/C01D50MSA7L/p1661255084098539)

**Comment by DaedalusG** [comment]
Starting work on a docker-compose solution here: https://github.com/sourcegraph/deploy-sourcegraph-docker/tree/openGrok-caddy-rewrites

Currently working for the directory use case in testing on my local machine
```
warrengifford@Warrens-MacBook-Pro ~ % curl http://localhost/source/xref/sourcegraph/client/web/src/enterprise/code-monitoring/testing/ -i
HTTP/1.1 302 Found
Location: github.com/sourcegraph/sourcegraph/-/blob/client/web/src/enterprise/code-monitoring/testing/
Server: Caddy
Date: Fri, 09 Sep 2022 19:29:36 GMT
Content-Length: 0
```

Will better deployment and customization options for our current caddy setup 👍

**Comment by DaedalusG** [comment]
Created a repo for exploring rewrite rules via the `nginx:stable` docker image to address this Issue:
https://github.com/sourcegraph/nginx-openGrok-poc/tree/main

TL;DR its likely in order to create a seamless translation of OpenGrok Searches to Sourcegraph Searches we'll likely need to explore a middleware in Sourcegraphs frontend service. There was some success with a strategy of tranlating openGrok page navigation to Sourcegraph searches, however, not enough data is provided in an OpenGrok Url to directly translate to a Sourcegraph Url (_openGrok contains no data on Codehost in its Url, whereas Sourcegraph does_)
